tests_new/integration_tests/core/stack/test_stack.py

```diff
@@ -0,0 +1,102 @@
+    assert_that(response).is_not_empty()
+
+
+def test_get_env_stack_logs_unauthorized(client2, session_env1):
+    assert_that(get_stack_logs).raises(GqlError).when_called_with(
+        client=client2,
+        target_uri=session_env1.environmentUri,
+        target_type='environment',
+    ).contains(
+        'UnauthorizedOperation',
+        'GET_ENVIRONMENT',
+        session_env1.environmentUri,
+    )
+
+
+def test_update_list_key_value_tags_add_tag(client1, session_env1, session_id):
+    # Test add tag on environment without tags
+    response = update_key_value_tags(
+        client1,
+        input={
+            'targetUri': session_env1.environmentUri,
+            'targetType': 'environment',
+            'tags': [
+                {'key': 'key1', 'value': session_id, 'cascade': False},
+                {'key': 'key2', 'value': session_id, 'cascade': True},
+            ],
+        },
+    )
+    assert_that(len(response)).is_equal_to(2)
+    assert_that(response[0]).contains_entry(key='key1', value=session_id, cascade=False)
+    assert_that(response[1]).contains_entry(key='key2', value=session_id, cascade=True)
+    # Test list tags after add
+    response = list_key_value_tags(client1, target_uri=session_env1.environmentUri, target_type='environment')
+    assert_that(len(response)).is_equal_to(2)
+    assert_that(response[0]).contains_entry(key='key1', value=session_id, cascade=False)
+    assert_that(response[1]).contains_entry(key='key2', value=session_id, cascade=True)
+
+
+def test_update_list_key_value_tags_add_tag_unauthorized(client2, session_env1, session_id):
+    assert_that(update_key_value_tags).raises(GqlError).when_called_with(
+        client=client2,
+        input={
+            'targetUri': session_env1.environmentUri,
+            'targetType': 'environment',
+            'tags': [
+                {'key': 'key1U', 'value': session_id, 'cascade': False},
+                {'key': 'key2U', 'value': session_id, 'cascade': True},
+            ],
+        },
+    ).contains(
+        'UnauthorizedOperation',
+        'UPDATE_ENVIRONMENT',
+        session_env1.environmentUri,
+    )
+
+
+def test_update_list_key_value_tags_add_tag_invalid_input(client1, session_env1, session_id):
+    assert_that(update_key_value_tags).raises(GqlError).when_called_with(
+        client=client1,
+        input={
+            'targetUri': session_env1.environmentUri,
+            'targetType': 'environment',
+            'tags': [
+                {'key': 'keyDuplicated', 'value': session_id, 'cascade': False},
+                {'key': 'keyDuplicated', 'value': session_id, 'cascade': True},
+            ],
+        },
+    ).contains(
+        'UnauthorizedOperation',
+        'SAVE_KEY_VALUE_TAGS',
+        'Duplicate tag keys found',
+    )
+
+
+def test_update_list_key_value_tags_delete_tags(client1, session_env1, session_id):
+    # Test delete tag
```

Comment:
should we check that key value tags are created on this environment to ensure they are being updated to be deleted? 

Either using a fixture to ensure tag is created to the environment or call the lines of code in `test_update_list_key_value_tags_add_tag` and `test_update_list_key_value_tags_delete_tags` in the same test to ensure this test is not dependent on others

Reply:
Yeep, let's do it!